tracing-opentelemetry/src/subscriber.rs

```diff
@@ -171,29 +171,27 @@ impl<'a> field::Visit for SpanEventVisitor<'a> {
 
 struct SpanAttributeVisitor<'a>(&'a mut otel::SpanBuilder);
 
+impl<'a> SpanAttributeVisitor<'a> {
+    fn record(&mut self, attribute: KeyValue) {
+        self.0.attributes.as_mut().map(|v| v.push(attribute));
```

Comment:
Hmm, this will silently do nothing if the `attributes` field is `None`. Since we set it to `Some` in `new_span`, we can expect this will always be the case, but this code doesn't currently *ensure* that.

Maybe it's worth adding a 
```suggestion
        debug_assert!(self.0.attributes.is_some());
        self.0.attributes.as_mut().map(|v| v.push(attribute));
```
or something? We could also just `expect` it to be `Some`, but maybe we shouldn't panic in production if an attribute wouldn't be recorded...